ui/partials/perf/comparechooserctrl.html

```diff
@@ -27,21 +27,23 @@
         <div class="panel-heading">New</div>
         <div class="panel-body">
           <label for="new-project-selector">Project</label>
-          <select id="new-project-selector" class="form-control" ng-model="newProject" ng-options="project.name for project in projects" ng-change="updateNewRevisionTips()"/>
+          <select id="new-project-selector" class="form-control" ng-model="newProject" ng-options="project.name for project in projects" ng-change="updateNewRevisionTips();getPreviousRevision()"/>
           <label for="new-revision-input">Revision</label>
             <div class="form-group" ng-class="{'has-error': newRevisionError}">
-            <input id="new-revision-input" maxlength="12" class="form-control" type="text" ng-model="newRevision" placeholder="Select or enter a revision"/>
+            <input id="new-revision-input" maxlength="12" class="form-control" type="text" ng-model="newRevision" placeholder="Select or enter a revision" ng-change="getPreviousRevision()"/>
             <div class="input-group-btn">
               <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
               Recent <span class="caret"></span></button>
               <ul class="dropdown-menu dropdown-menu-right" >
-                <li ng-repeat="tip in newTipList"><a ng-click="getNewTipRevision(tip.revision)">
+                <li ng-repeat="tip in newTipList"><a ng-click="getNewTipRevision(tip.revision);getPreviousRevision()">
                 {{tip.revision}} {{tip.author}}
                 </a></li>
               </ul>
             </div>
             </div>
             <label class="control-label" style="color:#A94442" ng-show="newRevisionError">{{newRevisionError}}"</label>
+            <a href="" ng-show="proposalRevision" ng-click="setProposalRevision()">Set the base revision to be the previous of this one.</a>
+            <img style="float:right" src="img/dancing_cat.gif" height="42" width="42" ng-show="proposalRevisionLoading"/>
```

Comment:
The dancing cat is cute but I think it's a bit confusing since it's not accompanied with a loading message. I think it's probably better to use an animated spinner, which is easy to do with font awesome:

https://fortawesome.github.io/Font-Awesome/examples/